### ISSUE TITLE
Block the whole page while ng-app was located in inner div

### ISSUE THREAD

**Issue by tan9** [McNull/angular-block-ui#91]
Our scenario is a bit complicated, and we want to use blockUI to block the whole page when needed.
However, our document contains 2 or more separate angular applications as below:

```html
<body>
  <div>an-app-which-will-be-manually-bootstrapped</div>
  <div ng-app="myApp"></div>
</body>
```

I found it is not easy to make blcokUI up properly. What will you suggest us on this scenario?

**Comment by tan9** [comment]
A dirty workaround we have is to detach app element while manually compile `<body block-ui="main"/>`:

```javascript
angular.module('myApp').run(function ($compile, $rootScope) {
    var body = angular.element("body");

    // detach ng-app element to prevent it from been compiled twice
    var app = body.find("[ng-app]");
    var appPrev = app.prev();
    app.detach();

    // manually compile block-ui and block-ui-container directives added by angular-block-ui
    $compile(body)($rootScope);

    // reattach ng-app
    appPrev.after(app);
});
```

**Comment by tan9** [comment]
Sorry, #62 suits our case very well.
